apps/web/src/components/Profile/NftGallery/Gallery.tsx

```diff
@@ -47,23 +62,88 @@ const Gallery: FC<Props> = ({ galleries }) => {
           }
         });
       }
-    } catch {}
+    } catch (error: any) {
+      toast.error(error?.messaage ?? ERROR_MESSAGE);
+    }
+  };
+
+  const setItemsToGallery = (items: Item[]) => {
+    setGallery({
+      ...gallery,
+      items,
+      isEdit: true,
+      toAdd: [],
+      toRemove: [],
+      alreadySelectedItems: items,
+      reArrangedItems: []
+    });
+  };
+
+  const onClickRearrange = () => {
+    const items = nfts.map((nft) => {
+      return { ...nft, itemId: `${nft.chainId}_${nft.contractAddress}_${nft.tokenId}` };
+    });
+    setIsRearrange(true);
+    setItemsToGallery(items);
+  };
+
+  const onClickEditGallery = () => {
+    setShowCreateModal(true);
+    const items = nfts.map((nft) => {
+      return { ...nft, itemId: `${nft.chainId}_${nft.contractAddress}_${nft.tokenId}` };
+    });
+    setItemsToGallery(items);
+  };
+
+  const onSaveRearrange = async () => {
+    try {
+      const updates = galleryStore.reArrangedItems?.map(
+        ({ tokenId, contractAddress, chainId, newOrder }, i) => {
+          return { tokenId, contractAddress, chainId, newOrder: newOrder ?? i };
+        }
+      );
+      await orderGallery({
+        variables: {
+          request: {
+            galleryId: galleryStore.id,
+            profileId: currentProfile?.id,
+            updates
+          }
+        }
+      });
+      const { data } = await fetchNftGalleries({
+        variables: { request: { profileId: currentProfile?.id } }
+      });
+      cache.modify({
+        fields: {
+          nftGalleries() {
+            cache.updateQuery({ query: NftGalleriesDocument }, () => ({
+              data: data?.nftGalleries as NftGallery[]
+            }));
+          }
+        }
+      });
```

Comment:
I think we can use cache inside `fetchNftGalleries` itself instead of using external cache hook